### PR TITLE
Formatting and fixing analyzer warnings

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

**Comment by nex3** [comment]
The formatting changes here seem uniformly as good as or worse than the existing formatting.

Also, I think we prefer Rietveld for dart-team-to-dart-team code reviews.

File: lib/src/version.dart [hunk 1]
```diff
@@ -86,15 +86,15 @@ class Version implements Comparable<Version>, VersionConstraint {
...
       : preRelease = preRelease == null ? [] : _splitParts(preRelease),
         build = build == null ? [] : _splitParts(build) {
-    if (major < 0) throw new ArgumentError(
-        'Major version must be non-negative.');
-    if (minor < 0) throw new ArgumentError(
-        'Minor version must be non-negative.');
-    if (patch < 0) throw new ArgumentError(
-        'Patch version must be non-negative.');
+    if (major <
+        0) throw new ArgumentError('Major version must be non-negative.');
```
Comment (jakemac53) [inline lib/src/version.dart]: ouch :(

**Comment by kevmoo** [comment]
A net-zero improvement is progress. It means that the next person who provides a patch can just format the code w/ tools and not worry about any one-offs.

Zero time spent arguing about indents in code reviews is a HUGE win.

File: lib/src/version.dart (hunk 1 above)
Comment (kevmoo) [inline lib/src/version.dart]: 100% Agree. Will file issue.

**Comment by jakemac53** [comment]
I agree with @kevmoo, anybody submitting a new pull request should be able to run the formatter on the entire folder they edited (or ideally the whole project) without getting extraneous edits from formatter pulled into their review.

**Comment by nex3** [comment]
I disagree. I generally care more about how readable code is versus how easy it is to write, and it's not that difficult to write decently-formatted code anyway. Until the formatter is strictly required for all code, I'm not in favor of running it on packages where it makes the formatting net worse.

**Comment by kevmoo** [comment]
Going with https://codereview.chromium.org/1109053002/

**Pull request closed without merging**